go/test/endtoend/vtgate/queries/derived/main_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
I suggest we move the E2E test located in `go/test/endtoend/vtgate/mysql80/derived` to this package.